libraries/standard/mqtt/src/mqtt.c

```diff
@@ -433,19 +437,37 @@ static int32_t recvExact( const MQTTContext_t * pContext,
                                pIndex,
                                bytesRemaining );
 
-        if( bytesRecvd >= 0 )
-        {
-            bytesRemaining -= ( size_t ) bytesRecvd;
-            totalBytesRecvd += ( int32_t ) bytesRecvd;
-            pIndex += bytesRecvd;
-        }
-        else
+        if( bytesRecvd < 0 )
         {
             LogError( ( "Network error while receiving packet: ReturnCode=%d",
                         bytesRecvd ) );
             totalBytesRecvd = bytesRecvd;
             receiveError = true;
         }
+        else if( bytesRecvd > ( int32_t ) bytesRemaining )
+        {
+            LogError( ( "Transport receive returned more bytes than expected: "
+                        "BytesExpected=%lu, BytesReceived=%d",
```

Comment:
Add a '.' at the end of logging message.

Reply:
Sure fixed everywhere it is missing in mqtt library.